blockchain/state_migration.go

```diff
@@ -97,7 +97,12 @@ func (bc *BlockChain) concurrentRead(db *statedb.Database, quitCh chan struct{},
 func (bc *BlockChain) migrateState(rootHash common.Hash) (returnErr error) {
 	bc.wg.Add(1)
 	defer func() {
-		bc.db.FinishStateMigration(returnErr == nil)
+		// If migration stops by quit signal, it doesn't finish migration and it it will restart again.
+		if returnErr != ErrQuitBySignal {
+			bc.mu.Lock()
+			bc.db.FinishStateMigration(returnErr == nil)
+			bc.mu.Unlock()
+		}
 		bc.wg.Done()
 	}()
 
```

Comment:
What do you think of resetting the return values of `StatusStateMigration` (bc.committedCnt, bc.pendingCnt, bc.progress, bc.migrationErr) ?
It could be confusing if a user started a migration and `StatusStateMigration` shows the previous status.

Reply:
I intended not to reset the information to check the result of state migration.
User may be confused. But for now, preserving information is helpful for us.

Reply:
Okay maybe later :)